README.md
<p align="center">
<img src = "https://raw.githubusercontent.com/ojasjain24/neon_widgets_flutter/master/demos/neon_widgets_logo.png" height = "300" width ="300"/>
</p>

<h1 align="center"> Neon Widgets💡 </h1>
 <h3 align="center"> Most commonly used flutter widgets in neon version.</h3>

[![License](https://img.shields.io/github/license/ojasjain24/neon_widgets_flutter?color=dark%20green)](https://github.com/ojasjain24/neon_widgets_flutter/blob/master/LICENSE)
[![Pub](https://img.shields.io/pub/v/neon_widgets?color=blue)](https://pub.dev/packages/neon_widgets)
[![GitHub stars](https://img.shields.io/github/stars/ojasjain24/neon_widgets_flutter?color=green&style=social)](https://github.com/ojasjain24/neon_widgets_flutter)

## Live Example : https://ojas-jain.web.app/#/


## Neon Widgets 2.0 => Breaking Changes

Animation disposal issues resolved. and every class' name have been changed, to comply with the standard dart syntax.

#### All the class names have been changed, to comply with the standard dart syntax. Previously every class had the name convention such that non-neon class' name started with the letter 'o', and neon classes started with 'oNeon'. But in the update, the neon class names now starts with 'Neon', and the letter 'o' has been removed from the starting of non-neon classes.

#### For example :
1 : 'oNeonContainer' is now changed to 'NeonContainer'.

2 : 'oNeonCustomVerticesProgressBar' is now changed to 'NeonCustomVerticesProgressBar'.

3 : 'oSearchBar' is now changed to 'SearchBar'.

## Getting started

Readme Changelog Installing Versions Scores Admin Activity log Use this package as a library Depend
on it Run this command:

Run this command:

With Dart:

```
 $ dart pub add neon_widgets
```

With Flutter:

```
 $ flutter pub add neon_widgets
```

This will add a line like this to your package's pubspec.yaml (and run an implicit dart pub get):

```
dependencies:
  neon_widgets: ^2.0.0
```

Alternatively, your editor might support dart pub get or flutter pub get. Check the docs for your
editor to learn more.

Import it Now in your Dart code, you can use:

```
import 'package:neon_widgets/neon_widgets.dart';
```

## Features

### ExampleApp

Neon Widgets :

<img src="https://raw.githubusercontent.com/ojasjain24/neon_widgets_flutter/master/demos/neonWidgets.gif" height="516" width="360">
<img src="https://raw.githubusercontent.com/ojasjain24/neon_widgets_flutter/master/demos/Screenshot%202022-02-21%20110732.png" height="580" width="360" >

Flickering Neon Widgets
(with random flicker rate) :

<img src="https://raw.githubusercontent.com/ojasjain24/neon_widgets_flutter/master/demos/flickerWidgetsDemo.gif" height="594" width="360">

Loaders :

 <img src = "https://raw.githubusercontent.com/ojasjain24/neon_widgets_flutter/master/demos/loaders.gif" />

### Widgets

neonText :

<img src = "https://user-images.githubusercontent.com/59116312/159339246-fda62119-75c2-4d5d-8f44-6c8f85db9800.gif"/>

```
// neon text

 NeonText(
   text: "Neon text",
   spreadColor: Colors.pink,
   blurRadius: 20,
   textSize: 74,
   textColor: Colors.white,
 ),

// flickering neon text

 FlickerNeonText(
   text: "Flicker neon text",
   flickerTimeInMilliSeconds: 600,
   spreadColor: Colors.blue,
   blurRadius: 20,
   textSize: 74,
 ),
```

neonSearchBar :

<img src="https://raw.githubusercontent.com/ojasjain24/neon_widgets_flutter/master/demos/Screenshot%202022-02-21%20110254.png"  >

```
NeonSearchBar()
```

Chat widgets :

<img src="https://raw.githubusercontent.com/ojasjain24/neon_widgets_flutter/master/demos/Screenshot%202022-02-21%20110417.png" >

```
// Left message card in neon theme
NeonLeftMsgCard(
  lightBlurRadius: 20,
  lightSpreadRadius: 5,
  msg:
      "Hi all, this is brand new library that provides most used widgets in neon and normal effect",
  time: "10:45",
),

const SizedBox(
  height: 20,
),

// Right message card in neon theme
NeonRightMsgCard(
    lightBlurRadius: 20,
    lightSpreadRadius: 10,
    backgroundColor: Colors.deepPurple,
    msg:
    "Hi all, this is brand new library that provides most used widgets in neon and normal effect",
    time: "10:45",
 ),

const SizedBox(
  height: 20,
),

// Left massage card

LeftMsgCard(
    msg:
    "Hi all, this is brand new library that provides most used widgets in neon and normal effect",
    time: "10:45",
),

const SizedBox(
  height: 20,
),

// Right massage card

RightMsgCard(
    msg:
    "Hi all, this is brand new library that provides most used widgets in neon and normal effect",
    time: "10:45",
),

const SizedBox(
  height: 20,
),

```

Add items widget :

<img src="https://raw.githubusercontent.com/ojasjain24/neon_widgets_flutter/master/demos/Screenshot%202022-02-21%20110455.png" >

```
Wrap(
direction: Axis.horizontal,
children: <Widget>[
  ...(searchedResults.map((e) =>
      NeonAddItemButton(
          data: e,
          borderColor: Colors.deepOrange.shade50,
          spreadColor: Colors.deepOrange,
          lightSpreadRadius: 3,
          lightBlurRadius: 18))),
],
),
```

NeonPoint widget :

<img src="https://raw.githubusercontent.com/ojasjain24/neon_widgets_flutter/master/demos/Screenshot%202022-02-21%20190238.png" >

```
NeonPoint(
  pointSize: 00,
  pointColor: Colors.red.shade100,
  spreadColor: Colors.red,
),
```

NeonLine widget :

<img src="https://raw.githubusercontent.com/ojasjain24/neon_widgets_flutter/master/demos/Screenshot%202022-02-21%20190619.png" >

```
NeonLine(
  spreadColor: Colors.brown,
  lightSpreadRadius: 30,
  lightBlurRadius: 90,
  lineWidth: 400,
  lineHeight: 0.02,
  lineColor: Colors.brown.shade100,
),
```

Flickering Neon widgets :

```
// Flickering neon line
FlickerNeonLine(
  spreadColor: Colors.brown,
  lightSpreadRadius: 30,
  lightBlurRadius: 60,
  lineWidth: 300,
  lineHeight: 2,
  lineColor: Colors.brown.shade100,

  // flicker rate which is selects a random value by default
  randomFlicker: false,
  flickerTimeInMilliSeconds: 1000,
),
```

```
// Flickering neon point
FlickerNeonPoint(
  pointSize: 010,
  pointColor: Colors.blue.shade100,
  spreadColor: Colors.blue,
),
```

```
// Flickering neon container
FlickerNeonContainer(
    spreadColor: Colors.green.shade700,
    child: WIDGET
),
```

Loaders :

```
 //loader type 1 : loader made of polynomial of n number of side
 NeonShapeVerticesProgressBar(number: 4)

 //loader type 2 : triangle shape with all colors customizable
 NeonTriangleVerticesProgressBar()

 //loader type 3 : square shape with all colors customizable
 NeonSquareVerticesProgressBar()

 //loader type 4 : image
 ImageProgressBar(image : "assets/images/abc.png")

```

Non Neon widgets :

<img src="https://raw.githubusercontent.com/ojasjain24/neon_widgets_flutter/master/demos/Screenshot%202022-02-21%20110545.png" >

```
SearchBar(),
const SizedBox(
  height: 20,
),
Wrap(
  direction: Axis.horizontal,
  children: <Widget>[
    ...(searchedResults
        .map((e) => AddItemButton(data: e))),
  ],
),
const SizedBox(
  height: 20,
),
Container(
  padding: const EdgeInsets.all(20),
  child: CircumscribingIconButton(
      icon: Icons.keyboard_arrow_left_sharp,
      onTap: () {},
      backgroundColor: Colors.purple),
),
```

## Usage

Example code provided

```
NeonContainer(
  spreadColor: Colors.green.shade700,
  child: CircumscribingIconButton(
        icon: Icons.arrow_right_alt_sharp,
        onTap: () {},
        backgroundColor: Colors.green.shade700),
    borderRadius: BorderRadius.circular(100),
    lightBlurRadius: 100,
    lightSpreadRadius: 50,
  borderColor: Colors.green.shade100
)
```

## Additional information

Wanna improve this package? Contribute to project on

github : https://github.com/ojasjain24/neon_widgets_flutter

create issue : https://github.com/ojasjain24/neon_widgets_flutter/issues

## Contributing rules :

1 : create an issue describing the feature.

2 : comment on this issue that you are interested in resolving this issue.

#### 3 : Create a Pull request in dev branch.



CHANGELOG.md
## 0.0.1

This package will provide you with most commonly used widgets with their normal and neon version.
There are multiple different types of widgets under this package, which can be used to create more
neon theme widget

## 0.0.2

Flickering widgets now available, with random and constant flicker intensity.

## 0.0.3

Loaders/ progressBars in neon theme.

## 0.0.4

readme updates

## 0.0.5

fixes

## 0.0.6

fixes

## 0.0.7

fixes

## 0.0.8

neon texts

## 0.0.9

readme updates

## 0.1.0

fix

## 0.1.1

neon text fixes

## 0.1.2

properties fixes

## 2.0.0
- Breaking changes : Class names changed to comply with the standard dart syntax. 
- Animation disposal issues resolved.


test/neon_widgets_flutter_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:neon_widgets/neon_widgets.dart';

void main() {
  test('adds one to input values', () {
    final calculator = Calculator();
    expect(calculator.addOne(2), 3);
    expect(calculator.addOne(-7), -6);
    expect(calculator.addOne(0), 1);
  });
}


example/exampleApp.dart
import 'package:flutter/material.dart';
import 'package:neon_widgets/neon_widgets.dart';

class exampleApp extends StatefulWidget {
  @override
  State<StatefulWidget> createState() {
    return exampleAppState();
  }
}

class exampleAppState extends State<exampleApp> {
  @override
  Widget build(BuildContext context) {
    List<String> searchedResults = [
      "Flutter",
      "PubDev",
      "Package",
      "Library",
      "Widgets",
      "New"
    ];

    return Scaffold(
      backgroundColor: Colors.black,
      appBar: oAppBar(
          context: context,
          heading: "Example App",
          iconColor: Colors.white,
          backgroundColor: Colors.deepPurple.shade600,
          onTap: () {},
          appBarShadowColor: Colors.purple),
      body: Stack(
        children: [
          SingleChildScrollView(
            child: Container(
              padding: const EdgeInsets.all(20),
              child: Column(
                children: [
                  const SizedBox(
                    height: 20,
                  ),
                  oNeonSearchBar(hint: "Search here..."),
                  const SizedBox(
                    height: 20,
                  ),
                  oNeonContainer(
                      spreadColor: Colors.teal.shade200,
                      borderColor: Colors.teal.shade50,
                      containerColor: Colors.black,
                      lightBlurRadius: 20,
                      lightSpreadRadius: 10,
                      borderRadius: BorderRadius.circular(10),
                      child: Padding(
                        padding: const EdgeInsets.all(10),
                        child: Column(
                          children: [
                            oNeonLeftMsgCard(
                                lightBlurRadius: 10,
                                lightSpreadRadius: 5,
                                msg:
                                    "Hi all, this is brand new library that provides most used widgets in neon and normal effect",
                                time: "10:45"),
                            const SizedBox(
                              height: 20,
                            ),
                            oNeonRightMsgCard(
                                lightBlurRadius: 20,
                                lightSpreadRadius: 10,
                                backgroundColor: Colors.deepPurple,
                                msg:
                                    "Hi all, this is brand new library that provides most used widgets in neon and normal effect",
                                time: "10:45"),
                            const SizedBox(
                              height: 20,
                            ),
                            oLeftMsgCard(
                                msg:
                                    "Hi all, this is brand new library that provides most used widgets in neon and normal effect",
                                time: "10:45"),
                            const SizedBox(
                              height: 20,
                            ),
                            oRightMsgCard(
                                msg:
                                    "Hi all, this is brand new library that provides most used widgets in neon and normal effect",
                                time: "10:45"),
                            const SizedBox(
                              height: 20,
                            ),
                            Wrap(
                              direction: Axis.horizontal,
                              children: <Widget>[
                                ...(searchedResults.map((e) =>
                                    oNeonAddItemButton(
                                        data: e,
                                        borderColor: Colors.deepOrange.shade50,
                                        spreadColor: Colors.deepOrange,
                                        lightSpreadRadius: 3,
                                        lightBlurRadius: 18))),
                              ],
                            ),
                            const SizedBox(
                              height: 20,
                            ),
                            Wrap(
                              direction: Axis.horizontal,
                              children: <Widget>[
                                ...(searchedResults.map((e) =>
                                    oNeonAddItemButton(
                                        data: e,
                                        borderColor: Colors.pink.shade50,
                                        spreadColor: Colors.pinkAccent,
                                        lightSpreadRadius: 3,
                                        lightBlurRadius: 18))),
                              ],
                            ),
                            const SizedBox(
                              height: 20,
                            ),
                            Wrap(
                              direction: Axis.horizontal,
                              children: <Widget>[
                                ...(searchedResults.map((e) =>
                                    oNeonAddItemButton(
                                        data: e,
                                        borderColor: Colors.blue.shade50,
                                        spreadColor: Colors.blue,
                                        lightSpreadRadius: 3,
                                        lightBlurRadius: 18))),
                              ],
                            ),
                            const SizedBox(
                              height: 20,
                            ),
                            oSearchBar(),
                            const SizedBox(
                              height: 20,
                            ),
                            Wrap(
                              direction: Axis.horizontal,
                              children: <Widget>[
                                ...(searchedResults
                                    .map((e) => oAddItemButton(data: e))),
                              ],
                            ),
                            const SizedBox(
                              height: 20,
                            ),
                            Container(
                              padding: const EdgeInsets.all(20),
                              child: oCircumscribingIconButton(
                                  icon: Icons.keyboard_arrow_left_sharp,
                                  onTap: () {},
                                  backgroundColor: Colors.purple),
                            ),
                            const SizedBox(
                              height: 100,
                            ),
                            oNeonPoint(
                              pointSize: 00,
                              pointColor: Colors.red.shade100,
                              spreadColor: Colors.red,
                            ),
                            const SizedBox(
                              height: 86.6,
                            ),
                            Row(
                              mainAxisAlignment: MainAxisAlignment.center,
                              children: [
                                oNeonPoint(
                                  pointSize: 00,
                                  pointColor: Colors.blue.shade100,
                                  spreadColor: Colors.blue,
                                ),
                                const SizedBox(
                                  width: 100,
                                ),
                                oNeonPoint(
                                  pointSize: 00,
                                  pointColor: Colors.green.shade100,
                                  spreadColor: Colors.green,
                                ),
                              ],
                            ),
                            const SizedBox(
                              height: 400,
                            ),
                            oNeonLine(
                              spreadColor: Colors.brown,
                              lightSpreadRadius: 30,
                              lightBlurRadius: 90,
                              lineWidth: 400,
                              lineHeight: 0.02,
                              lineColor: Colors.brown.shade100,
                            ),
                            const SizedBox(
                              height: 100,
                            ),
                          ],
                        ),
                      )),
                ],
              ),
            ),
          ),
          Container(
            padding: const EdgeInsets.all(20),
            alignment: Alignment.bottomRight,
            child: oNeonContainer(
                spreadColor: Colors.green.shade700,
                child: oCircumscribingIconButton(
                    icon: Icons.arrow_right_alt_sharp,
                    onTap: () {},
                    backgroundColor: Colors.green.shade700),
                borderRadius: BorderRadius.circular(100),
                lightBlurRadius: 100,
                lightSpreadRadius: 50,
                borderColor: Colors.green.shade100),
          ),
        ],
      ),
    );
  }
}


example/example_app.dart
import 'package:flutter/material.dart';
import 'package:neon_widgets/neon_widgets.dart';

class ExampleApp extends StatefulWidget {
  const ExampleApp({Key? key}) : super(key: key);

  @override
  State<StatefulWidget> createState() {
    return ExampleAppState();
  }
}

class ExampleAppState extends State<ExampleApp> {
  @override
  Widget build(BuildContext context) {
    List<String> searchedResults = [
      "Flutter",
      "PubDev",
      "Package",
      "Library",
      "Widgets",
      "New"
    ];

    return Scaffold(
      backgroundColor: Colors.black,
      appBar: oAppBar(
          context: context,
          heading: "Example App",
          iconColor: Colors.white,
          backgroundColor: Colors.deepPurple.shade600,
          onTap: () {},
          appBarShadowColor: Colors.purple),
      body: Stack(
        children: [
          SingleChildScrollView(
            child: Container(
              padding: const EdgeInsets.all(20),
              child: Column(
                children: [
                  const SizedBox(
                    height: 20,
                  ),
                  const NeonSearchBar(hint: "Search here..."),
                  const SizedBox(
                    height: 20,
                  ),
                  NeonContainer(
                      spreadColor: Colors.teal.shade200,
                      borderColor: Colors.teal.shade50,
                      containerColor: Colors.black,
                      lightBlurRadius: 20,
                      lightSpreadRadius: 10,
                      borderRadius: BorderRadius.circular(10),
                      child: Padding(
                        padding: const EdgeInsets.all(10),
                        child: Column(
                          children: [
                            const NeonLeftMsgCard(
                                lightBlurRadius: 10,
                                lightSpreadRadius: 5,
                                msg:
                                    "Hi all, this is brand new library that provides most used widgets in neon and normal effect",
                                time: "10:45"),
                            const SizedBox(
                              height: 20,
                            ),
                            const NeonRightMsgCard(
                                lightBlurRadius: 20,
                                lightSpreadRadius: 10,
                                backgroundColor: Colors.deepPurple,
                                msg:
                                    "Hi all, this is brand new library that provides most used widgets in neon and normal effect",
                                time: "10:45"),
                            const SizedBox(
                              height: 20,
                            ),
                            const LeftMsgCard(
                                msg:
                                    "Hi all, this is brand new library that provides most used widgets in neon and normal effect",
                                time: "10:45"),
                            const SizedBox(
                              height: 20,
                            ),
                            const RightMsgCard(
                                msg:
                                    "Hi all, this is brand new library that provides most used widgets in neon and normal effect",
                                time: "10:45"),
                            const SizedBox(
                              height: 20,
                            ),
                            Wrap(
                              direction: Axis.horizontal,
                              children: <Widget>[
                                ...(searchedResults.map((e) =>
                                    NeonAddItemButton(
                                        data: e,
                                        borderColor: Colors.deepOrange.shade50,
                                        spreadColor: Colors.deepOrange,
                                        lightSpreadRadius: 3,
                                        lightBlurRadius: 18))),
                              ],
                            ),
                            const SizedBox(
                              height: 20,
                            ),
                            Wrap(
                              direction: Axis.horizontal,
                              children: <Widget>[
                                ...(searchedResults.map((e) =>
                                    NeonAddItemButton(
                                        data: e,
                                        borderColor: Colors.pink.shade50,
                                        spreadColor: Colors.pinkAccent,
                                        lightSpreadRadius: 3,
                                        lightBlurRadius: 18))),
                              ],
                            ),
                            const SizedBox(
                              height: 20,
                            ),
                            Wrap(
                              direction: Axis.horizontal,
                              children: <Widget>[
                                ...(searchedResults.map((e) =>
                                    NeonAddItemButton(
                                        data: e,
                                        borderColor: Colors.blue.shade50,
                                        spreadColor: Colors.blue,
                                        lightSpreadRadius: 3,
                                        lightBlurRadius: 18))),
                              ],
                            ),
                            const SizedBox(
                              height: 20,
                            ),
                            const SearchBar(),
                            const SizedBox(
                              height: 20,
                            ),
                            Wrap(
                              direction: Axis.horizontal,
                              children: <Widget>[
                                ...(searchedResults
                                    .map((e) => AddItemButton(data: e))),
                              ],
                            ),
                            const SizedBox(
                              height: 20,
                            ),
                            Container(
                              padding: const EdgeInsets.all(20),
                              child: CircumscribingIconButton(
                                  icon: Icons.keyboard_arrow_left_sharp,
                                  onTap: () {},
                                  backgroundColor: Colors.purple),
                            ),
                            const SizedBox(
                              height: 100,
                            ),
                            NeonPoint(
                              pointSize: 00,
                              pointColor: Colors.red.shade100,
                              spreadColor: Colors.red,
                            ),
                            const SizedBox(
                              height: 86.6,
                            ),
                            Row(
                              mainAxisAlignment: MainAxisAlignment.center,
                              children: [
                                NeonPoint(
                                  pointSize: 00,
                                  pointColor: Colors.blue.shade100,
                                  spreadColor: Colors.blue,
                                ),
                                const SizedBox(
                                  width: 100,
                                ),
                                NeonPoint(
                                  pointSize: 00,
                                  pointColor: Colors.green.shade100,
                                  spreadColor: Colors.green,
                                ),
                              ],
                            ),
                            const SizedBox(
                              height: 400,
                            ),
                            NeonLine(
                              spreadColor: Colors.brown,
                              lightSpreadRadius: 30,
                              lightBlurRadius: 90,
                              lineWidth: 400,
                              lineHeight: 0.02,
                              lineColor: Colors.brown.shade100,
                            ),
                            const SizedBox(
                              height: 100,
                            ),
                          ],
                        ),
                      )),
                ],
              ),
            ),
          ),
          Container(
            padding: const EdgeInsets.all(20),
            alignment: Alignment.bottomRight,
            child: NeonContainer(
                spreadColor: Colors.green.shade700,
                borderRadius: BorderRadius.circular(100),
                lightBlurRadius: 100,
                lightSpreadRadius: 50,
                borderColor: Colors.green.shade100,
                child: CircumscribingIconButton(
                    icon: Icons.arrow_right_alt_sharp,
                    onTap: () {},
                    backgroundColor: Colors.green.shade700)),
          ),
        ],
      ),
    );
  }
}
